internal/controller/kokumetricsconfig_controller.go

```diff
@@ -409,6 +410,10 @@ func (r *MetricsConfigReconciler) setAuthentication(ctx context.Context, authCon
 	}
 
 	if cr.Status.Authentication.AuthType == metricscfgv1beta1.Basic {
+		// set basic auth deprecation message
+		deprecationMsg := "Basic authentication is deprecated and will be removed after December 31, 2024. Please switch to service-account authentication."
+		cr.Status.Authentication.DeprecationNotice = deprecationMsg
```

Comment:
This is likely compiled away, but there really isn't any point in putting this string into a variable:
```suggestion
		cr.Status.Authentication.DeprecationNotice = "Basic authentication is deprecated and will be removed after December 31, 2024. Please switch to service-account authentication."
```

Reply:
that's a fair point